src/net.cpp

```diff
@@ -1737,8 +1737,10 @@ void StartNode(boost::thread_group& threadGroup, CScheduler& scheduler)
         CAddrDB adb;
         if (adb.Read(addrman))
             LogPrintf("Loaded %i addresses from peers.dat  %dms\n", addrman.size(), GetTimeMillis() - nStart);
-        else
+        else {
+            addrman.Clear(); // Addrman can be in an inconsistent state after failure, reset it
             LogPrintf("Invalid or missing peers.dat; recreating\n");
+        }
```

Comment:
shouldn't we actually recreate peers.dat, with `DumpAddresses`, now that addrman has been cleared?

Reply:
added the `DumpAddresses` call, and did the same for when the banlist.dat file is missing too.